### DISCUSSION TITLE
Can sensor submitted job not appear as ad_hoc materializations

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18376] [Q&A]
is there a way to make sensor submitted jobs *NOT* appear as ad_hoc materializations

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1701280731200739?thread_ts=1701280731.200739&cid=C01U954MEER).*

**Comment by nickvazz** [comment]
![image](https://github.com/dagster-io/dagster/assets/66751111/af602b81-8d5f-4263-86d1-e3f9aca28e1f)

and instead show up as the job that was submitted? and maybe a backfill specific section?

**Reply by smackesey** [reply]
Would you mind posting some code? I just tried this example and the runs did not show up as Ad hoc materializations, but rather as submitted jobs:

```
### 231204-sensorjob.py

from dagster import (
    asset,
    Definitions,
    AssetExecutionContext,
    sensor,
)
from dagster._core.definitions.run_request import RunRequest
from dagster._core.definitions.unresolved_asset_job_definition import define_asset_job

@asset
def foo():
    return 1

@asset
def bar():
    return 1

@asset
def baz():
    return 1

my_asset_job = define_asset_job("my_asset_job", [foo, bar, baz])

@sensor(job=my_asset_job)
def my_sensor(context):
    return RunRequest()


defs = Definitions(
    assets=[foo, bar, baz],
    jobs=[my_asset_job],
    sensors=[my_sensor],
)
```

**Reply by nickvazz** [reply]
Is it possible because I return within my sensor a list of partitioned `RunRequest`s?

This is roughly what I do using a sensor factory for a subset of assets

```python
def _sensor(context: SensorEvaluationContext):   
    run_requests = []        
    asset_key = asset.key.path
    partitions = context.instance.get_dynamic_partitions("/".join(asset_key))
    cursor = json.loads(context.cursor) if context.cursor is not None else {}
    cursor |= {partition: 0 for partition in partitions if partition not in cursor}

    for partition in partitions:
        last_mtime = get_float_formatted_date(cursor[partition])

        if some_logic_checking_if_file_has_been_modified:
            run_key = f'{"__".join(asset_key)}_{partition}:{last_mtime}'
            run_requests.append(RunRequest(run_key=run_key, partition_key=partition))
   
    if not any(run_requests):
        return SensorResult(skip_reason="Found no new runs", cursor=json.dumps(cursor))

    return SensorResult(run_requests=run_requests, cursor=json.dumps(cursor), )
```

**Reply by smackesey** [reply]
Hmm are your sensors not being defined with a job at all, but instead with just an `asset_selection`? If so, then it makes sense that they show up as "ad hoc materializations". If you want them to show up as a job run you should define an asset job with the target assets using `define_asset_job` and run that instead.
